crates/router/src/connector/utils.rs

```diff
@@ -319,6 +319,47 @@ impl RefundsRequestData for types::RefundsData {
     }
 }
 
+#[derive(Clone, Debug, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GooglePayWalletData {
+    #[serde(rename = "type")]
+    pub pm_type: String,
+    pub description: String,
+    pub info: GooglePayPaymentMethodInfo,
+    pub tokenization_data: GpayTokenizationData,
+}
+
+#[derive(Clone, Debug, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GooglePayPaymentMethodInfo {
+    pub card_network: String,
+    pub card_details: String,
+}
+
+#[derive(Clone, Debug, serde::Serialize)]
+pub struct GpayTokenizationData {
+    #[serde(rename = "type")]
+    pub token_type: String,
+    pub token: String,
+}
+
+impl From<api_models::payments::GooglePayWalletData> for GooglePayWalletData {
+    fn from(data: api_models::payments::GooglePayWalletData) -> Self {
```

Comment:
Both structs have the same name. Some developer integrating Googlepay might be confused as to which type to use and why this conversion is necessary, can you please explain as to why we have to do it like this?

Reply:
All connectors requires the googlepay_data in camelcase but we serialize and deserialize it in snake_case, so we initially made serialization in camelcase and deserialization in snake_case previously. Now an issue happened when they stored the gpay token in camelcase in vault and tried retrieving in snake_case, so to avoid this issue we revert back to old mechanism of all snake_case concept. now whatever connector that requires the google token strictly in camelcase we add a custom logic for those to convert them to camelcase. That's why both have same name but are in different crates

Reply:
Thanks for the explanation.